osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/domain/TypedSecret.java

```diff
@@ -8,29 +8,20 @@
  */
 package org.opensmartgridplatform.secretmanagement.application.domain;
 
+import lombok.Getter;
+
 /**
  * TypedSecret stores a secret (not necessarily an encrypted secret), along with it's type.
  */
+@Getter
 public class TypedSecret {
-    private SecretType secretType;
-    private String secret;
-
-    public String getSecret() {
-        return this.secret;
-    }
-
-    public SecretType getSecretType() {
-        return this.secretType;
-    }
+    private final SecretType secretType;
+    private final byte[] secret;
 
-    public void setSecret(final String secret) {
+    public TypedSecret(final byte[] secret, final SecretType secretType) {
```

Comment:
I am not entirely sure what is needed here and how far you would go to keep this safe, but given there are only getters it looks like the intention is that the internals are not to be modified.
Did you consider cloning/copying the `secret` (setting it in the constructor, and returning it from the getter) and requiring `secretType` to be non-null?

(Even if not needed immediately in the long run you may run into cases where you had made your utility data types unmodifiable.)